#96 The create command should not overwrite a running game

Purlieu opened the pull request:
Only Create Games when In progress in Game.  I think this is correct.  I couldn't think of where else this needed to be checked.  Under CLI, it uses an exception to check if the factory will create.

holyzala commented on project/eScavenge/Game.py:
```diff
@@ -280,8 +280,10 @@ def __init__(self, maker):
     def create_game(self, *args, **kwargs):
-        return self.maker(*args, **kwargs)
-
+        if self.game._Game__started:
```
self in this case is GameFactory, which has no game member. This condition should always cause an exception.

Purlieu replied:
@holyzala Would I be able to fix that by just using Game.started?  instead of self.game.Game__started.  From what I understand in regards to python, calling Game.stared is the class variable shared among all instances of that class.

holyzala replied:
No, at this point we can not be sure there even is a Game in existence. And to check it would require a CLI instance passed down somehow. This check should be done in CLI I think, but definitely not here in the factory.

holyzala commented:
The exception that you mention in CLI actually does absolutely nothing. It's just copy/paste code from other functions that use args. We never index anything in that call, so it can't ever return an IndexError.

Purlieu commented:
@holyzala Okay I think my confusion came in checking somehow in game, but that makes more sense checking in CLI which I assume is just a simple fix that I updated.   

As far as the exception goes, I switched it to a ValueError which I assume is relevant.

holyzala commented on project/eScavenge/CLI.py:
```diff
@@ -128,9 +128,11 @@ def end(self, _):
 @need_admin
 def create(self, _):
+    if self.game.started or not self.game.ended:
+        return Errors.GAME_IN_PROGRESS
```
We need to return a string here, not an error.

holyzala commented on project/eScavenge/CLI.py:
```diff
@@ -128,9 +128,11 @@ def end(self, _):
     try:
         self.game = GameFactory(make_game).create_game()
-    except IndexError:
+    except ValueError:
```
ValueError is also meaningless I think. I think the try/except can go away completely, there aren't any exceptions that can be thrown that I know of...

Purlieu replied:
Shouldn't there be some kind of indicator of whether or not we are getting a game created or not though? I was perusing the exception list and couldn't find anything relevant.  If that's the case then we should change the start and end commands as well as they are currently using the indexerror exception as well

holyzala replied:
GameFactory cannot return anything except a new game. So if it completes successfully then we will get a new game. There are currently no cases in our code that will prevent it from running successfully. So the only possible exceptions we could get are completely random system errors (out of memory, system unstable, and the like) that would kill every other call we try and make most likely. So there's nothing for us to catch.
You are probably correct on the start and end, since we don't pass anything to them there's nothing (that we introduce) that could cause an exception.

holyzala replied:
For try/excepts in that location the intent was to catch int("abc") [ValueError] when the user puts in a string and we expect an int. or args[i] when i doesn't exist [IndexError] when the user misses a parameter.

Purlieu commented:
@holyzala I removed the exceptions from the 3 commands.   Since we only get a new game from Game Factory, if we do the checks to make sure there is no current game running, then it will always return successful if it reaches the line containing the Game Factory right?

Purlieu commented:
I gotta look into it.  It's throwing errors on all fronts.
